django/views/generic/dates.py

```diff
@@ -485,10 +485,14 @@ def get_dated_items(self):
 
         date_field = self.get_date_field()
         week_format = self.get_week_format()
-        week_start = {
-            '%W': '1',
-            '%U': '0',
-        }[week_format]
+        try:
+            week_start = {
+                '%W': '1',
+                '%U': '0',
+            }[week_format]
+        except KeyError:
+            raise ValueError("unknown week format: %s" % week_format)
```

Comment:
Use single quotes. Capitalized "unknown". Add "Choices are: ...".

Reply:
@timgraham Thank you for the comment. changes are done.